services/teamcity/teamcity-build.service.js

```diff
@@ -3,7 +3,7 @@
 const Joi = require('joi')
 const TeamCityBase = require('./teamcity-base')
 
-const buildStatusTextRegex = /^Success|Failure|Error|Tests( failed: \d+( \(\d+ new\))?)?(,)?( passed: \d+)?(,)?( ignored: \d+)?(,)?( muted: \d+)?$/
+const buildStatusTextRegex = /^Success|Failure|Error|Tests( failed: \d+( \(\d+ new\))?)?(,)?( passed: \d+)?(,)?( ignored: \d+)?(,)?( muted: \d+)?(.*)?$/
```

Comment:
I think removing the `$` should be equivalent:

```suggestion
const buildStatusTextRegex = /^Success|Failure|Error|Tests( failed: \d+( \(\d+ new\))?)?(,)?( passed: \d+)?(,)?( ignored: \d+)?(,)?( muted: \d+)?/
```

Also could you add a note about the issue this is solving (and a link)?

Reply:
https://github.com/badges/shields/issues/3244 
here you go

Reply:
Ah, thanks! I meant that it'd be good for @calebcartwright to include an explanation in the code of why the isn't matching the entire string.